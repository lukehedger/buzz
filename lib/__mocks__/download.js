import * as request from '../util/request'
import makeSwarmURL from '../util/makeSwarmURL'

const download = (hash, provider, schemeType = 'LONG') => {
  if (!provider) {
    throw new Error(
      'Swarm node provider not set. Use `buzz.setProvider` to set the Swarm provider.'
    )
  }

  return request
    .get(makeSwarmURL(provider, 'raw', schemeType, hash))
    .then(res => res.text())
}

export default download
