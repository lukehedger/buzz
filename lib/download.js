import * as request from './util/request'
import makeSwarmURL from './util/makeSwarmURL'

/**
 * Download data from a Swarm content hash
 *
 * @param  {String} hash                Swarm content hash
 * @param  {String} provider            Swarm node provider
 * @param  {String} [schemeType='LONG'] Swarm URI scheme type
 * @return {String}                     Downloaded data
 */
const download = (hash, provider, schemeType = 'LONG') => {
  if (!provider) {
    throw new Error(
      'Swarm node provider not set. Use `buzz.setProvider` to set the Swarm provider.'
    )
  }

  return request
    .get(makeSwarmURL(provider, 'raw', schemeType, hash))
    .then(res => res.text())
}

export default download
