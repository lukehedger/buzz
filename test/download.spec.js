import download from '../lib/download'

jest.mock('../lib/download')
jest.mock('../lib/util/request')

const SWARM_PROVIDER = 'http://localhost:8500'

describe('buzz :: download', () => {
  it('Should download content at Swarm hash', async () => {
    const actual = await download(
      'f070f4f87c0018a4026ff2c14895e9ef525e81c4cf4d768e1bf658e570927a13',
      SWARM_PROVIDER
    )

    expect(actual).toMatchSnapshot()
  })

  it('Should throw an error if Swarm provider not set', () => {
    const actual = () => download('buzz')

    expect(actual).toThrowErrorMatchingSnapshot()
  })
})
