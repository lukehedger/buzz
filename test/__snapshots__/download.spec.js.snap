// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buzz :: download Should download content at Swarm hash 1`] = `"buzz"`;

exports[`buzz :: download Should throw an error if Swarm provider not set 1`] = `"Swarm node provider not set. Use \`buzz.setProvider\` to set the Swarm provider."`;
